js/service.js

```diff
@@ -1,11 +1,11 @@
 import env from "../config/env.generated.json";
+import BluebirdPromise from "bluebird";
```

Comment:
We can probably just import `bluebird` as `Promise`. Since other functions in this file use the build-in `Promise` and I don't want to introduce more changes to this already-huge PR, I'm importing `bluebird` as `BluebirdPromise` for now. We can file follow-up PR to clean up this file.